subprojects/core/src/main/java/org/gradle/api/Script.java

```diff
@@ -359,4 +360,14 @@
      */
     ResourceHandler getResources();
 
+    /**
+     * Creates a {@code ConfigurablePropertyState} implementation based on the provided class.
+     *
+     * @param clazz The class to be used for property state.
+     * @return The property state. Never returns null.
+     * @throws org.gradle.api.InvalidUserDataException If the provided class is null.
+     * @since 3.5
+     */
+    @Incubating
+    <T> ConfigurablePropertyState<T> property(Class<T> clazz);
```

Comment:
As discussed we now have a method for creating a `Provider` and a `PropertyState`.